### PR TITLE
Xpub getters for generalistic descriptors

### PR DESCRIPTION
**Pull request by darosior**
The new typesystem, which is awesome, allows us to do this safely now :)

### CODE REVIEW

File: src/scripts.rs [hunk 1]
```diff
@@ -579,6 +624,27 @@ impl CpfpDescriptor {
 
         Ok(CpfpDescriptor(cpfp_descriptor!(managers)))
     }
+
+    /// Get all the xpubs used in this Unvault descriptor.
```
Comment (danielabrozzoni) [inline src/scripts.rs:628]: Should be Cpfp

File: src/scripts.rs [hunk 2]
```diff
@@ -579,6 +624,27 @@ impl CpfpDescriptor {
...
+    pub fn xpubs(&self) -> Vec<DescriptorPublicKey> {
+        let ms = match self.0 {
+            Descriptor::Wsh(ref wsh) => match wsh.as_inner() {
+                WshInner::Ms(ms) => ms,
+                WshInner::SortedMulti(_) => {
+                    unreachable!("Deposit descriptor is not a sorted multi")
```
Comment (danielabrozzoni) [inline src/scripts.rs:634]: Should be Cpfp

File: src/scripts.rs [hunk 3]
```diff
@@ -579,6 +624,27 @@ impl CpfpDescriptor {
...
+                }
+            },
+            _ => unreachable!("Deposit descriptor is always a P2WSH"),
```
Comment (danielabrozzoni) [inline src/scripts.rs:637]: Same

Comment (darosior) [reply src/scripts.rs:637]: copypasta :/, thanks

**Comment by danielabrozzoni** [comment]
utACK 8e3c4053f674bf20699ff6e72eebf2b0f0f6e93c

**Pull request merged**